### PR TITLE
fixed compile issues and warnings for iOS platform

### PR DESCRIPTION
**Pull request by kgbook**
fixed issue: `exec_program` is deprecated since version 3.0, CMake >= 3.28 prefer that this command never be called and the `NEW` behavior for this policy is to issue a `FATAL_ERROR` when the command is called.

fixed warning: cmake policy CMP0054.

update .gitignore: ignore files generated by CLion and CMake.

### CODE REVIEW

**Comment by maxsharabayko** [comment]
CMake 2.8.12 supports `execute_proces` 👍 
Wonder why Travis build fails.

**Comment by ethouris** [comment]
But `execute_process` call syntax is possible with different parameters. Not every case of call syntax is supported on the old version.

**Comment by kgbook** [comment]
> CMake 2.8.12 supports `execute_proces` 👍 Wonder why Travis build fails.

I have no idea why Travis-CI build fails. I tried to get the job logs from travis-ci, but unable to access the job logs.  It said:
> "We're sorry, but this data is not available. Please check the repository settings in Travis CI." 

If the travis-ci job log and CMake log are available, I would be happy to assist in identifying and resolving the issue.

**Comment by ethouris** [comment]
All I can see is that there's some initial cmake checks that finally gets "errors ocurred!" by completely no visible reason.

Rest of the things are post-factum and some general error report from a command.

**Comment by maxsharabayko** [comment]
Right, not much insight, except that obviously CMake fails somewhere.
```
...
$ if [ "$TRAVIS_COMPILER" == "x86_64-w64-mingw32-g++" ]; then export CC="x86_64-w64-mingw32-gcc"; export CXX="x86_64-w64-mingw32-g++"; cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE $BUILD_OPTS -DENABLE_UNITTESTS="OFF" -DUSE_OPENSSL_PC="OFF" -DOPENSSL_ROOT_DIR="$PWD/openssl" -DCMAKE_SYSTEM_NAME="Windows"; elif [ "$TRAVIS_OS_NAME" == "linux" ]; then cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE $BUILD_OPTS -DENABLE_UNITTESTS="ON"; elif [ "$TRAVIS_OS_NAME" == "osx" ]; then export PKG_CONFIG_PATH=$(brew --prefix openssl)"/lib/pkgconfig"; cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE $BUILD_OPTS -DENABLE_UNITTESTS="ON"; fi
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/g++
-- Check for working CXX compiler: /usr/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring incomplete, errors occurred!
See also "/home/travis/build/Haivision/srt/CMakeFiles/CMakeOutput.log".
...
```

**Comment by kgbook** [comment]
> Right, not much insight, except that obviously CMake fails somewhere.
> 
> ```
> ...
> $ if [ "$TRAVIS_COMPILER" == "x86_64-w64-mingw32-g++" ]; then export CC="x86_64-w64-mingw32-gcc"; export CXX="x86_64-w64-mingw32-g++"; cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE $BUILD_OPTS -DENABLE_UNITTESTS="OFF" -DUSE_OPENSSL_PC="OFF" -DOPENSSL_ROOT_DIR="$PWD/openssl" -DCMAKE_SYSTEM_NAME="Windows"; elif [ "$TRAVIS_OS_NAME" == "linux" ]; then cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE $BUILD_OPTS -DENABLE_UNITTESTS="ON"; elif [ "$TRAVIS_OS_NAME" == "osx" ]; then export PKG_CONFIG_PATH=$(brew --prefix openssl)"/lib/pkgconfig"; cmake . -DCMAKE_BUILD_TYPE=$BUILD_TYPE $BUILD_OPTS -DENABLE_UNITTESTS="ON"; fi
> -- The C compiler identification is GNU 5.4.0
> -- The CXX compiler identification is GNU 5.4.0
> -- Check for working C compiler: /usr/bin/gcc
> -- Check for working C compiler: /usr/bin/gcc -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler: /usr/bin/g++
> -- Check for working CXX compiler: /usr/bin/g++ -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Configuring incomplete, errors occurred!
> See also "/home/travis/build/Haivision/srt/CMakeFiles/CMakeOutput.log".
> ...
> ```

See also "/home/travis/build/Haivision/srt/CMakeFiles/CMakeOutput.log".

**Comment by ethouris** [comment]
That's this problem with these machines. You can only run some script afterwards that will extract some information from the files, but you can't access files through any shell or something.

**Comment by maxsharabayko** [comment]
cmake version 3.5.1-1ubuntu3.

> See also "/home/travis/build/Haivision/srt/CMakeFiles/CMakeOutput.log".

I'm afraid it is not available. What can be done is to echo the contents to the terminal in this PR.

**Comment by kgbook** [comment]
> But `execute_process` call syntax is possible with different parameters. Not every case of call syntax is supported on the old version.

The `COMMAND` and `OUTPUT_VARIABLE` options are supported in CMake version 3.0, as mentioned in the [CMake v3.0 execute_process documentation](https://cmake.org/cmake/help/v3.0/command/execute_process.html). 

No documents available for cmake version older than v3.0. While I have tested this functionality on CMake version 3.28, I cannot confirm its compatibility with CMake version 2.8.12.

**Comment by ethouris** [comment]
There are likely also other `execute_process` calls in the same file and if they work, their syntax should be followed.

**Comment by kgbook** [comment]
> There are likely also other `execute_process` calls in the same file and if they work, their syntax should be followed.

I have downloaded CMake version 2.8.12 from [CMake release v2.8.12.2](https://github.com/Kitware/CMake/archive/refs/tags/v2.8.12.2.tar.gz). Upon inspecting the `CMakeLists.txt` file at line 405, the syntax for the `execute_process` command appears to be the same, with `COMMAND` and `OUTPUT_VARIABLE` options supported.

```cmake
execute_process(COMMAND ${CMAKE_CXX_COMPILER}
     ${CMAKE_CXX_COMPILER_ARG1} -dumpversion
     OUTPUT_VARIABLE _GXX_VERSION
   )
```

This suggests that the syntax remains consistent even in CMake version 2.8.12.

**Comment by ethouris** [comment]
So, likely you have to modify the Travis execution configuration by checking the error from cmake command and dump its log file on the screen.

**Comment by maxsharabayko** [comment]
Right, the syntax seems to be the same. `exec_program` was already deprecated back then.
https://cmake.org/cmake/help/v2.8.12/cmake.html#command:execute_process

**Comment by maxsharabayko** [comment]
I suppose `VERSION_GREATER_EQUAL` was not there.
New in version 3.7.

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,10 @@ else()
 	project(SRT VERSION ${SRT_VERSION} LANGUAGES C CXX)
 endif()
 
+if (${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.28.1")
```
Comment (maxsharabayko) [inline CMakeLists.txt:27]: VERSION_GREATER_EQUAL was added in CMake 3.7/

**Comment by kgbook** [comment]
travis-ci build pass now.

**Review by maxsharabayko — APPROVED** [review]

**Comment by maxsharabayko** [comment]
@kgbook Thanks a lot for this PR!

**Pull request merged**